### ISSUE TITLE
`defaultValue` broken when `multiple: false`

### ISSUE THREAD

**Issue by maxramqvist** [jeffwcx/inquirer-select-pro#9]
First of all; Fantastic prompt - thanks for the great work!

I found a bug, but I'm not sure how to fix it. Currently when `multiple` is set to `false` the `defaultValue` cant be selected, the returned value is `null`.

New file for reproducing `src/examples/reproduce.ts`:
```typescript
import { select } from '../src/index';
import { top100Films } from '../tests/data';

const selected = await select({
  message: "What's your favorite movie?",
  options: top100Films,
  defaultValue: top100Films[1].value,
  multiple: false,
});
if (selected === null) {
  throw new Error('No movie selected'); // this happens if I select the default value (`The Godfather (1972)`) from the list.
}

console.log(`Selected: ${selected}`);
```

When multiple: true is set - it works as expected.

Selecting any other value than the defaultValue works.

Edit:

While trying to see if I could fix this I found that whatever I select, except for the `defaultValue` which becomes `null`, the `defaultValue` becomes the selected option.

**Comment by maxramqvist** [comment]
I've found a way to fix it on my end - but I'm not sure if it makes sense for the bigger picture or architecture you have in mind.

Replacing line [163-173](https://github.com/jeffwcx/inquirer-select-pro/blob/302c5c26fbc8a4137f392c996f6ff12d03ab2340/src/useSelect.ts#L163) with this works on my end. 
```typescript
      if (targetOption.checked) {
        const currentSelection = multiple
          ? selections.current.filter(
              (op) => !equals(targetOption.value, op.value),
            )
          : (selections.current = [{ ...targetOption }])
        setBehavior('deselect', true)
        selections.current = currentSelection
      } else {
        setBehavior('select', true)
        if (multiple) {
          selections.current = [...selections.current, { ...targetOption }]
        } else {
          selections.current = [{ ...targetOption }]
        }
      }
    ```
    
If it makes sense I can do a PR - of if you'd rather add it yourself?

**Comment by jeffwcx** [comment]
This bug was fixed in [version 1.0.0-alpha.8](https://www.npmjs.com/package/inquirer-select-pro).
